kernel-community/signature-economies
# Code restructure + Arweave upload

The 3 major components of the application, the way I'm thinking are:

- **Highlight**
- Free sign
- Slider mint

For this PR i have been completely focussed on Highlight (the `/src/components/HighlightModal`), the functionality that I tested:

- highlighting 
- clicking mint
- uploading to arweave
    - for arweave, to test locally:
    - run a local `arlocal` node: `$ npx arlocal`
    - either generate a new wallet using `$ node src/utils/arweave/tasks/account/wallet` or fund the test account in `src/utils/arweave/key.json` using `$ node src/utils/arweave/tasks/account/fund` (funds 100 AR)
    - check balance of the account stored (in key.json) using `$ node src/utils/arweave/tasks/account/balance`
    - nothing needs to be changed in the code since the arweave config is currently set for `localhost:1984`
    - to mine a block run `$ node src/utils/arweave/tasks/mine`
    - to switch to arweave mainnet, change config in `$ node src/utils/arweave/config.json`
    - the function just logs the transaction id in browser console as soon as the upload finishes, navigating to `http://localhost:1984/${id}` should display the image
    - <img width="1388" alt="Screenshot 2022-05-16 at 3 44 03 AM" src="https://user-images.githubusercontent.com/10832414/168541043-75ad97d6-9cd2-4476-8f80-bed2d3793600.png">



In the code, this PR:
- adds routing
- uses HighlightContext to maintain state for the modal
- adds some tweaks/fixes

What's left :
- [ ] Generating metadata
- [ ] Minting the NFT 
- [ ] The image that is being uploaded is just the components generated using p5js, does not include the text (need to figure this out later)
